content/en/docs/concepts/configuration/secret.md

```diff
@@ -61,1189 +116,1167 @@ merged into the `data` field. If a key appears in both the `data` and the
-To consume all keys from the secret, all of them must be listed in the `items` field.
-All listed keys must exist in the corresponding secret. Otherwise, the volume is not created.
-
-#### Secret files permissions
-
-You can set the file access permission bits for a single Secret key.
-If you don't specify any permissions, `0644` is used by default.
-You can also set a default mode for the entire Secret volume and override per key if needed.
+Create the Secret:
+```shell
+kubectl apply -f mysecret.yaml
+```
 
-For example, you can specify a default mode like this:
+Use `envFrom` to define all of the Secret's data as container environment variables. The key from the Secret becomes the environment variable name in the Pod.
 
 ```yaml
 apiVersion: v1
 kind: Pod
 metadata:
-  name: mypod
+  name: secret-test-pod
 spec:
   containers:
-  - name: mypod
-    image: redis
-    volumeMounts:
-    - name: foo
-      mountPath: "/etc/foo"
-  volumes:
-  - name: foo
-    secret:
-      secretName: mysecret
-      defaultMode: 0400
+    - name: test-container
+      image: k8s.gcr.io/busybox
+      command: [ "/bin/sh", "-c", "env" ]
+      envFrom:
+      - secretRef:
+          name: mysecret
+  restartPolicy: Never
 ```
 
-Then, the secret will be mounted on `/etc/foo` and all the files created by the
-secret volume mount will have permission `0400`.
+### Use case: Pod with SSH keys
 
-Note that the JSON spec doesn't support octal notation, so use the value 256 for
-0400 permissions. If you use YAML instead of JSON for the Pod, you can use octal
-notation to specify permissions in a more natural way.
+Create a Secret containing some SSH keys:
 
-Note if you `kubectl exec` into the Pod, you need to follow the symlink to find
-the expected file mode. For example,
-
-Check the secrets file mode on the pod.
+```shell
+kubectl create secret generic ssh-key-secret --from-file=ssh-privatekey=/path/to/.ssh/id_rsa --from-file=ssh-publickey=/path/to/.ssh/id_rsa.pub
 ```
-kubectl exec mypod -it sh
 
-cd /etc/foo
-ls -l
-```
+The output is similar to:
 
-The output is similar to this:
 ```
-total 0
-lrwxrwxrwx 1 root root 15 May 18 00:18 password -> ..data/password
-lrwxrwxrwx 1 root root 15 May 18 00:18 username -> ..data/username
+secret "ssh-key-secret" created
 ```
 
-Follow the symlink to find the correct file mode.
+You can also create a `kustomization.yaml` with a `secretGenerator` field containing ssh keys.
 
-```
-cd /etc/foo/..data
-ls -l
-```
+{{< caution >}}
+Think carefully before sending your own SSH keys: other users of the cluster may have access
+to the Secret.
 
-The output is similar to this:
-```
-total 8
--r-------- 1 root root 12 May 18 00:18 password
--r-------- 1 root root  5 May 18 00:18 username
-```
+You could instead create an SSH private key representing a service identity that you want to be
+accessible to all the users with whom you share the Kubernetes cluster, and that you can revoke
```

Comment:
nit: This sentence could be broken up into several sentences.

You could instead create an SSH private key representing a service identity that is accessible to all users in your Kubernetes cluster. If the credentials are compromised, you can revoke the access to ...